mlflow/utils/conda.py

```diff
@@ -84,12 +85,43 @@ def _get_conda_executable_for_create_env():
     return conda_env_create_path
 
 
-def _list_conda_environments():
-    prc = process._exec_cmd([get_conda_bin_executable("conda"), "env", "list", "--json"])
+def _list_conda_environments(extra_env=None):
+    prc = process._exec_cmd(
+        [get_conda_bin_executable("conda"), "env", "list", "--json"], extra_env=extra_env
+    )
     return list(map(os.path.basename, json.loads(prc.stdout).get("envs", [])))
 
 
-def get_or_create_conda_env(conda_env_path, env_id=None, capture_output=False):
+def _get_conda_extra_envs(env_root_dir=None):
+    if env_root_dir is not None:
+        # Create isolated conda package cache dir "conda_pkgs" under the env_root_dir
+        # for each python process.
+        # Note: shared conda package cache dir causes race condition issues:
+        # See https://github.com/conda/conda/issues/8870
+        # See https://docs.conda.io/projects/conda/en/latest/user-guide/configuration/use-condarc.html#specify-environment-directories-envs-dirs
+        # and https://docs.conda.io/projects/conda/en/latest/user-guide/configuration/use-condarc.html#specify-package-directories-pkgs-dirs
+
+        conda_env_path = os.path.join(env_root_dir, "conda_envs")
+        conda_pkgs_path = os.path.join(env_root_dir, "conda_pkgs")
```

Comment:
```suggestion
        conda_pkgs_path = os.path.join(env_root_dir, "conda_pkgs")
```

Can we include `cache` in the directory name?